azure/src/test/java/org/apache/iceberg/azure/AzurePropertiesTest.java

```diff
@@ -90,4 +93,34 @@ public void testNoConnectionString() {
     props.applyClientConfiguration("account", clientBuilder);
     verify(clientBuilder).endpoint("https://account");
   }
+
+  @Test
+  public void testSharedKe() {
+    Assertions.assertThatIllegalStateException()
+        .isThrownBy(
+            () ->
+                new AzureProperties(
+                    ImmutableMap.of(
+                        ADLS_SHARED_KEY_ACCOUNT_KEY, "not-really-base64-encoded-key-here")))
+        .withMessage(
+            String.format(
+                "Azure authentication: shared-key requires both %s and %s",
+                ADLS_SHARED_KEY_ACCOUNT_NAME, ADLS_SHARED_KEY_ACCOUNT_KEY));
+    Assertions.assertThatIllegalStateException()
+        .isThrownBy(
+            () -> new AzureProperties(ImmutableMap.of(ADLS_SHARED_KEY_ACCOUNT_NAME, "account")))
+        .withMessage(
+            String.format(
+                "Azure authentication: shared-key requires both %s and %s",
+                ADLS_SHARED_KEY_ACCOUNT_NAME, ADLS_SHARED_KEY_ACCOUNT_KEY));
+    Assertions.assertThatNoException()
```

Comment:
rather than checking that no exception has been thrown, it would be better to make sure that the right credentials have been configured (similar to how other tests in this class do it): `verify(clientBuilder).credential(...)`